sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConsumerClient.cs

```diff
@@ -611,123 +608,105 @@ public virtual async Task CloseAsync(CancellationToken cancellationToken = defau
+        ///   <paramref name="channel" /> in the background, using a dedicated transport consumer
+        ///   instance.
         /// </summary>
         ///
-        /// <param name="eventHubName">The name of the event hub with the subscription; used for informational purposes only.</param>
-        /// <param name="partitionId">The identifier for the partition to associate with the subscription; used for informational purposes only.</param>
-        /// <param name="consumerGroup">The name of the consumer group to associate with the subscription; used for informational purposes only.</param>
-        /// <param name="maximumQueuedEvents">The maximum number of events to queue while waiting for them to be read.</param>
-        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to signal the request to cancel the event enumeration.</param>
+        /// <param name="partitionId">The identifier of the partition from which events should be read.</param>
+        /// <param name="startingPosition">The position within the partition's event stream that reading should begin from.</param>
+        /// <param name="channel">The channel to which events should be published.</param>
+        /// <param name="publishingCancellationSource">A cancellation source which can be used for signaling publication to stop.</param>
         ///
-        /// <returns>The elements of a channel subscription, including its identifier and a reader for the subscribed channel.</returns>
+        /// <returns>A function to invoke when publishing should stop; once complete, background publishing is no longer taking place and he <paramref name="channel"/> has been marked as final.</returns>
         ///
-        private (Guid Identifier, ChannelReader<EventData> ChannelReader) SubscribeToChannel(string eventHubName,
-                                                                                             string partitionId,
-                                                                                             string consumerGroup,
-                                                                                             int maximumQueuedEvents,
-                                                                                             CancellationToken cancellationToken)
+        /// <remarks>
+        ///   This method assumes co-ownership of the <paramref name="channel" />, marking its writer as completed
+        ///   when publishing is complete or when an exception is encountered.
+        ///
+        ///   This method also assumes co-ownership of the <paramref name="publishingCancellationSource" /> and will request cancellation
+        ///   from it when publishing is complete or when an exception is encountered.
+        /// </remarks>
+        ///
+        private async Task<Func<Task>> PublishPartitionEventsToChannelAsync(string partitionId,
```

Comment:
This abstraction is intended to be shared by `ReadEventsFromPartition` and `ReadEvents` _(pending)_.